src/foam/dao/JDBCPooledDataSource.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.X;
+import org.apache.commons.dbcp2.*;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+import javax.sql.DataSource;
+import java.sql.DriverManager;
+
+public class JDBCPooledDataSource {
+
+  protected static String SQL_POOL_NAME = "SQLPool";
+
+  // Holds a reference to the connection pool ( .getConnection() )
+  protected static DataSource dataSource_;
```

Comment:
Why is this static?

Reply:
the dataSource_ object holds a pool of connections so I made it static so that all the users of the class use the same pool